src/library/smartshape.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+$module SmartShape
+]]
+local smartshape = {}
+
+
+--[[
+% smartshape_entrybased
+
+Creates an entry based SmartShape based on two input notes. If a type is not specified, creates a slur.
+
+@ start_note (FCNoteEntry) Starting note for SmartShape.
+@ end_note (FCNoteEntry) Ending note for SmartShape.
+@ type (SMARTSHAPE_TYPES or string)
+]]
+function smartshape.add_entry_based_smartshape(start_note, end_note, shape_type)
+    local smartshape = finale.FCSmartShape()
+    smartshape:SetEntryAttachedFlags(true)
+    -- Folks: Let me know if this is just too much... or not enough...
+    local types = {"slurauto" = finale.SMARTSHAPE_SLURAUTO, "slur_auto" = finale.SMARTSHAPE_SLURAUTO, "autoslur" = finale.SMARTSHAPE_SLURAUTO, "auto_slur" = finale.SMARTSHAPE_SLURAUTO, 
```

Comment:
Totally fine with me that there's an absurd number of items. For this, though, I'd probably move it to the top-level so the table doesn't need to be reconstructed for every single function invocation. I'd also put one item per line.

```lua
local types = {
    "slurauto" = finale.SMARTSHAPE_SLURAUTO,
    "slur_auto" = finale.SMARTSHAPE_SLURAUTO,
    "autoslur" = finale.SMARTSHAPE_SLURAUTO,
    "auto_slur" = finale.SMARTSHAPE_SLURAUTO,
    -- ...
}
```